### PR TITLE
[ISSUE #3652] refactored the code, extracted a new method and used switch

### LINKED ISSUE #3652: [Enhancement] Code reuse is extracted as a method.[SendMessageBatchV2ProtocolResolver]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

image
located at:
eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/http/SendMessageBatchV2ProtocolResolver.java line 67-91,94-118

### Describe the solution you'd like

image
image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by dipankr**
Fixes #3652.

### Motivation

*there were duplicate code in the file, refactored the duplicate code to a method and refactored further to make the code more readable*

### Modifications

*refactored the code, extracted a new method and used switch to make code more readable*

### Documentation

- Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/http/SendMessageBatchV2ProtocolResolver.java [hunk 1]
```diff
@@ -42,84 +42,50 @@ public static CloudEvent buildEvent(Header header, Body body) throws ProtocolHan
...
-                    .withExtension(ProtocolKey.PROTOCOL_DESC, protocolDesc)
-                    .withExtension(ProtocolKey.PROTOCOL_VERSION, protocolVersion)
-                    .withExtension(SendMessageBatchV2RequestBody.BIZSEQNO, sendMessageBatchV2RequestBody.getBizSeqNo())
-                    .withExtension(SendMessageBatchV2RequestBody.PRODUCERGROUP, sendMessageBatchV2RequestBody.getProducerGroup())
-                    .withExtension(SendMessageBatchV2RequestBody.TTL, sendMessageBatchV2RequestBody.getTtl());
-                if (StringUtils.isNotEmpty(sendMessageBatchV2RequestBody.getTag())) {
-                    cloudEventBuilder = cloudEventBuilder.withExtension(SendMessageRequestBody.TAG, sendMessageBatchV2RequestBody.getTag());
-                }
-                event = cloudEventBuilder.build();
+            switch (parse(sendMessageBatchV2RequestHeader.getProtocolVersion())) {
```
Comment (Alonexc) [inline eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/http/SendMessageBatchV2ProtocolResolver.java:46]: SpecVersion.parse(sendMessageBatchV2RequestHeader.getProtocolVersion())

Comment (dipankr) [reply eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/http/SendMessageBatchV2ProtocolResolver.java]: can you help with this build error:
image

Comment (Alonexc) [reply eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/http/SendMessageBatchV2ProtocolResolver.java]: image
I was able to compile successfully here.

Comment (dipankr) [reply eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/http/SendMessageBatchV2ProtocolResolver.java]: I haven't had a green build ever in my local. it fails every single time. I doubt it is something to do with the versions. I have set the java to 1.8 in local, can you help me with what else can I do to get it working?

Comment (dipankr) [reply eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/http/SendMessageBatchV2ProtocolResolver.java]: I was finally able to get a green build. The issue was timeout error when downloading jars
fix:
added 
-Dorg.gradle.internal.http.connectionTimeout=120000 -Dorg.gradle.internal.http.socketTimeout=120000

to org.gradle.jvmargs in gradle.properties

**Review by Alonexc — APPROVED** [review]

**Review by xwm1992 — APPROVED** [review]

**Pull request merged**